server/src-lib/Hasura/GraphQL/Validate.hs

```diff
@@ -172,12 +171,16 @@ validateGQ (QueryParts opDef opRoot fragDefsL varValsM) = do
     G.OperationTypeQuery -> return $ RQuery selSet
     G.OperationTypeMutation -> return $ RMutation selSet
     G.OperationTypeSubscription ->
-      case Seq.viewl selSet of
-        Seq.EmptyL     -> throw500 "empty selset for subscription"
-        fld Seq.:< rst -> do
-          unless (null rst) $
-            throwVE "subscription must select only one top level field"
-          return $ RSubscription fld
+      case selSet of
+        Seq.Empty       -> throw500 "empty selset for subscription"
+        (_ Seq.:<| rst) -> do
+          -- As an internal testing feature, we support subscribing to
+          -- multiple selection sets.  First check if the
+          -- corresponding directive is set.
+          let multipleAllowed = elem (G.Directive "_multiple_subscriptions" []) (G._todDirectives opDef)
```

Comment:
I would probably call it, `_multiple_top_level_fields`, your call though, I'm not particular about this. Formatting of the comment is as per the style guide (wrap it at 80 characters)?

Reply:
Formatting of the comment is as per Emacs' `M-q` :smile:.

I'll fix the formatting and rename the directive to `_multiple_top_level_fields`.